lib/worker.js

```diff
@@ -53,3 +56,42 @@ if (!module.parent) {
   // HTTPS
   if (config.ssl.listen !== false) https.createServer(config.ssl, app).listen(config.ssl.listen || 8443);
 }
+
+// And start a cleanup loop over the work directory; all workers will run the same loop which is ok - deleting
+// orphaned in-flight work is idempotent and we'll
+
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+// Scan for dead files on an interval between 5 and 10 minutes (randomized to minimize overlap with other workers).
+var cleanupMs = (5 + getRandomInt(0, 5)) * 60 * 1000;
+
+// Expire working files that are 10 minutes old or older.
+var maxAgeMs = 10 * 60 * 1000;
+
+setInterval(function() {
+  var now = new Date().getTime();
+  fs.readdir(config.work_dir, function(err, files) {
+    if (err) {
+      log.error('Work dir cleanup failed to readdir: ' + err);
+      return;
+    }
+    files.forEach(function(file) {
+      var workingFile = path.join(config.work_dir, file);
+      fs.stat(workingFile, function(err, stat) {
+        if (err) {
+          log.error('Failed to stat file ' + file + ': ' + err);
+        }
+        var ctime = new Date(stat.ctime).getTime();
+        if (now - ctime > maxAgeMs) {
+          fs.unlink(workingFile, function(err) {
+            if (err) {
+              log.info('Failed to delete expired working file ' + workingFile + ', now=' + now + ', ctime=' + ctime + ', err=' + err);
+            }
+          });
```

Comment:
Should we log when files get cleaned up/ unlinked as part of this process? It seems like it would be a rare event but something that we might be interested in knowing if something fails in the future.